ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/PackageConfigurationFactory.kt

```diff
@@ -9,34 +9,150 @@ import com.revenuecat.purchases.ui.revenuecatui.helpers.Logger
+                                first = firstPackage,
+                                default = defaultPackage,
+                                all = packageInfos,
+                            ),
+                        ),
+                    ),
+                )
+            }
+            PackageConfigurationType.MULTITIER -> { this
+                val tiers = paywallData.config.tiers ?: run {
+                    return Result.failure(PackageConfigurationError("No tier found for $packageIdsInConfig"))
+                }
+
+                val (locale, localizedConfigurationByTier) = paywallData.tieredLocalizedConfiguration
+
+                val all = tiers.associateWith { tier ->
+                    val localizationForTier = localizedConfigurationByTier[tier.id]
+                        ?: return Result.failure(PackageConfigurationError("No localization found for ${tier.id}"))
+
+                    val packageInfosForTier = reprocessPackagesForTiers(
+                        from = availablePackages,
+                        filter = tier.packages,
+                        localization = localizationForTier,
+                        variableDataProvider = variableDataProvider,
+                        activelySubscribedProductIdentifiers = activelySubscribedProductIdentifiers,
+                        nonSubscriptionProductIdentifiers = nonSubscriptionProductIdentifiers,
+                        locale = locale,
+                    )
+
+                    val firstPackage = packageInfosForTier.firstOrNull()
+                        ?: return Result.failure(PackageConfigurationError("No packages found for tier ${tier.id}"))
+                    val defaultPackage = packageInfosForTier
+                        .firstOrNull { it.rcPackage.identifier == tier.defaultPackage } ?: firstPackage
+
+                    TemplateConfiguration.PackageConfiguration.MultiPackage(
+                        first = firstPackage,
+                        default = defaultPackage,
+                        all = packageInfosForTier,
+                    )
+                }
+
+                val allTierInfos = all.entries.map { (tier, packageInfo) ->
+                    val tierName = packageInfo.default.localization.tierName
+                        ?: return Result.failure(
+                            PackageConfigurationError("No localized tier name found for ${tier.id}"),
+                        )
+
+                    TemplateConfiguration.TierInfo(
+                        id = tier.id,
+                        name = tierName,
+                        defaultPackage = packageInfo.default,
+                        packages = packageInfo.all,
+                    )
+                }
+
+                Result.success(
+                    Pair(
+                        locale,
+                        TemplateConfiguration.PackageConfiguration.MultiTier(
+                            firstTier = allTierInfos.first(),
```

Comment:
Hmm maybe we want to deal with this later, but it would be good to have a way to specify the initial tier from the dashboard?

Reply:
I agree with this! I'll add it to the schema and then we can add a dashboard setting for it later 😇